integr8ly/cloud-resource-operator
# deleting bundle network resources on deletion of last rds or elasticache

## Overview

Jira: https://issues.redhat.com/browse/INTLY-5160

Pushing an early pr for review to verify that the approach is good. 

## Verification

- Install cloud-resource-operator with bundled resources
  - git checkout v0.16.1 
  - `make cluster/prepare NAMESPACE=cloud-resource-operator-<name>`
  - `make cluster/seed/managed/redis NAMESPACE=clourd-resource-operator-<name>`
  - `make cluster/seed/managed/redis NAMESPACE=cloud-resource-operator-<name>`
  - `make run/local NAMESPACE=cloud-resource-operator-<name>`
  - Get the cluster vpc id by running `aws ec2 describe-vpcs` and noting the vpc id with the tag     ```{
                    "Key": "kubernetes.io/cluster/<clustername>-<randomchars>",
                    "Value": "owned"
                }```
  - Wait for the resources to install. The resources of interest are below and will follow naming syntax `<clustername><randomchars>subnetgroup` and `<clustername><randomchars>securitygroup` and an associated vpc noted above.
      - ec2 security group with security group 
      - rds subent group with the name similar to the below
      - elasticache subnet group with the name similar to the below

See images below for examples for cluster mw-colab-byoc - using ui rather than cli as there's problems with the cli for some of the resources.
<img width="270" alt="Screenshot 2020-07-07 at 15 25 10" src="https://user-images.githubusercontent.com/6498727/86796110-2947f400-c066-11ea-9441-d601240c21a7.png">
<img width="352" alt="Screenshot 2020-07-07 at 15 26 47" src="https://user-images.githubusercontent.com/6498727/86796260-51375780-c066-11ea-94d5-067c7ef99e32.png">
       - elasticache subnet group with the name similar to the below
<img width="605" alt="Screenshot 2020-07-07 at 15 27 46" src="https://user-images.githubusercontent.com/6498727/86796380-6f04bc80-c066-11ea-81b4-fb02dfba78f1.png">


- Run the operator with the deletion changes in place - i.e. use this branch
  - `git checkout INTLY-5160`
  - `make run NAMESPACE=cloud-resource-operator-<name>`
  - delete either a postgres or redis
  - wait for the resource and the cr to delete (this will take a few minutes as the postgres is deleted)
  - verify that the network resources (rds subnet group, elasticache group & ec2 security group) are still in place
      - check in the aws console as above
  - delete the remaining either postgres or redis
  - verify that the operator doesn't error on reconcile if any of the resources are removed.
  - verify that the network resources are deleted - verify in the aws console the resources listed above are deleted.
